pkg/apis/serving/v1alpha1/llm_inference_service_types.go

```diff
@@ -0,0 +1,304 @@
+	// +optional
+	Router *RouterSpec `json:"router,omitempty"`
+
+	// Prefill configuration for disaggregated serving.
+	// When this section is included, the controller creates a separate deployment for prompt processing (prefill)
+	// in addition to the main 'decode' deployment, inspired by the llm-d architecture.
+	// This allows for independent scaling and hardware allocation for prefill and decode steps.
+	// +optional
+	Prefill *WorkloadSpec `json:"prefill,omitempty"`
+
+	// BaseRefs allows inheriting and overriding configurations from one or more LLMInferenceServiceConfig instances.
+	// The controller merges these base configurations, with the current LLMInferenceService spec taking the highest precedence.
+	// When multiple baseRefs are provided, the last one in the list overrides previous ones.
+	// +optional
+	BaseRefs []corev1.LocalObjectReference `json:"baseRefs,omitempty"`
+}
+
+// WorkloadSpec defines the configuration for a deployment workload, such as replicas and pod specifications.
+type WorkloadSpec struct {
+	// Number of replicas for the deployment.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Parallelism configurations for the runtime, such as tensor and pipeline parallelism.
+	// These values are used to configure the underlying inference runtime (e.g., vLLM).
+	// +optional
+	Parallelism *ParallelismSpec `json:"parallelism,omitempty"`
+
+	// Template for the main pod spec.
+	// In a multi-node deployment, this configures the "head" or "master" pod.
+	// In a disaggregated deployment, this configures the "decode" pod if it's the top-level template,
+	// or the "prefill" pod if it's within the Prefill block.
+	// +optional
+	Template *corev1.PodSpec `json:"template,omitempty"`
+
+	// Worker configuration for multi-node deployments.
+	// The presence of this field triggers the creation of a multi-node (distributed) setup.
+	// This spec defines the configuration for the worker pods, while the main 'Template' field defines the head pod.
+	// The controller is responsible for enabling discovery between head and worker pods.
+	// +optional
+	Worker *corev1.PodSpec `json:"worker,omitempty"`
+}
+
+// LLMModelSpec defines the model source and its characteristics.
+type LLMModelSpec struct {
+	// URI of the model, specifying its location, e.g., hf://meta-llama/Llama-4-Scout-17B-16E-Instruct
+	// The storage-initializer init container uses this URI to download the model.
+	URI apis.URL `json:"uri"`
+
+	// Name is the name of the model as it will be set in the "model" parameter for an incoming request.
+	// If omitted, it will default to `metadata.name`. For LoRA adapters, this field is required.
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// Criticality defines how important it is to serve the model compared to other models.
+	// This is used by the Inference Gateway scheduler.
+	// +optional
+	Criticality *igwapi.Criticality `json:"criticality,omitempty"`
```

Comment:
why not priority which is a more familiar concept.

Reply:
I think it's better to use the name from the inferenceGateway to prevent confusion.